ViZDoomPPO/load_model_generate_dataset.py

```diff
@@ -422,73 +421,149 @@ def make_gif(agent, file_path, num_episodes=1):
+        if os.path.isdir(episode_path):
+            for pkl_file in os.listdir(episode_path):
+                if pkl_file.endswith('.pkl'):
+                    with open(os.path.join(episode_path, pkl_file), 'rb') as f:
+                        data = pickle.load(f)
+                        batch.append(data)
+
+                    if len(batch) >= batch_size:
+                        df = pd.DataFrame(batch)
+                        table = pa.Table.from_pandas(df, schema=schema)
+
+                        if writer is None:
+                            writer = pq.ParquetWriter(output_parquet, schema)
+
+                        writer.write_table(table)
+                        batch = []
+
+    # Write any remaining data
+    if batch:
+        df = pd.DataFrame(batch)
+        table = pa.Table.from_pandas(df, schema=schema)
+        if writer is None:
+            writer = pq.ParquetWriter(output_parquet, schema)
+        writer.write_table(table)
+
+    if writer:
+        writer.close()
+
+    print(f"Data saved to {output_parquet}")
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Generate GIF or Parquet file from pretrained PPO agent")
+    parser.add_argument("--output", choices=["gif", "parquet"], required=True, help="Output format")
+    parser.add_argument("--episodes", type=int, default=1, help="Number of episodes to run")
+    parser.add_argument("--upload", action="store_true", help="Upload the output to Hugging Face Hub")
+    parser.add_argument("--hf_token", help="Hugging Face API token")
+    parser.add_argument("--hf_repo", help="Hugging Face repository name")
+    return parser.parse_args()
+
+MODEL_PATH = "logs/models/dm_longrun_test/2point5mil_iter_skip2.zip"
+
+def zip_directory(directory_path, zip_path):
+    with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
+        for root, _, files in os.walk(directory_path):
+            for file in files:
+                file_path = os.path.join(root, file)
+                arcname = os.path.relpath(file_path, directory_path)
+                zipf.write(file_path, arcname)
+
+def upload_to_hf(local_path, repo_id, token):
+    api = HfApi()
+    try:
+        # Ensure the dataset repository exists
+        api.create_repo(repo_id=repo_id, token=token, repo_type="dataset", exist_ok=True)
+        
+        # Upload the zip file
+        api.upload_file(
+            path_or_fileobj=local_path,
+            path_in_repo=os.path.basename(local_path),
+            repo_id=repo_id,
+            repo_type="dataset",
+            token=token,
+        )
+        print(f"Uploaded {local_path} to dataset {repo_id}")
+    except Exception as e:
+        print(f"Error uploading file: {e}")
 
 if __name__ == "__main__":
+    args = parse_arguments()
+
     scenario = "deathmatch_simple"
 
     env_args = {
         "scenario": scenario,
-        "frame_skip": 2,
+        "frame_skip": 1,
```

Comment:
Shouldn't that be 4 instead? Should probably be a constant stored in a config file

Reply:
For training, frame skip 4 allows better convergence, but in evaluation I think you always want frame skip 1.